test/integration/scoped_rds_integration_test.cc

```diff
@@ -159,34 +172,46 @@ route_configuration_name: foo_route1
 )EOF";
   const std::string scope_route2 = R"EOF(
 name: foo_scope2
-route_configuration_name: foo_route2
+route_configuration_name: foo_route1
 key:
   fragments:
-    - string_key: x-foo-key
+    - string_key: x-bar-key
 )EOF";
 
-  on_server_init_function_ = [this, &scope_route1, &scope_route2]() {
+  const std::string route_config_tmpl = R"EOF(
+      name: {}
+      virtual_hosts:
+      - name: integration
+        domains: ["*"]
+        routes:
+        - match: {{ prefix: "/" }}
+          route: {{ cluster: {} }}
+)EOF";
+
+  on_server_init_function_ = [&]() {
     createScopedRdsStream();
     sendScopedRdsResponse({scope_route1, scope_route2}, "1");
+    createRdsStream();
+    sendRdsResponse(fmt::format(route_config_tmpl, "foo_route1", "cluster_foo_1"), "1");
+    sendRdsResponse(fmt::format(route_config_tmpl, "foo_route1", "cluster_foo_2"), "2");
   };
   initialize();
-
-  test_server_->waitForCounterGe("http.config_test.scoped_rds.foo-scoped-routes.update_attempt", 1);
+  test_server_->waitForCounterGe("http.config_test.scoped_rds.foo-scoped-routes.update_attempt", 2);
```

Comment:
What is the reason for two update attempts at this point?

Reply:
it implies "one success" + "another attempt to update fired to management server".
When a subscription is spun up, the attempt number is increased by 1 already before any response sent back from management server.

Reply:
thanks, got it.